x/slashing/keeper.go

```diff
@@ -101,11 +102,19 @@ func (k Keeper) handleValidatorSignature(ctx sdk.Context, pubkey crypto.PubKey,
 	}
 	minHeight := signInfo.StartHeight + k.SignedBlocksWindow(ctx)
 	if height > minHeight && signInfo.SignedBlocksCounter < k.MinSignedPerWindow(ctx) {
-		// Downtime confirmed, slash, revoke, and jail the validator
-		logger.Info(fmt.Sprintf("Validator %s past min height of %d and below signed blocks threshold of %d", pubkey.Address(), minHeight, k.MinSignedPerWindow(ctx)))
-		k.validatorSet.Slash(ctx, pubkey, height, power, k.SlashFractionDowntime(ctx))
-		k.validatorSet.Revoke(ctx, pubkey)
-		signInfo.JailedUntil = ctx.BlockHeader().Time + k.DowntimeUnbondDuration(ctx)
+		validator := k.validatorSet.ValidatorByPubKey(ctx, pubkey)
+		if validator != nil && !validator.GetRevoked() {
```

Comment:
shouldn't we be checking this sooner? we are we even bothering fetching and updating the signInfo if he's no longer a validator ?

Reply:
Hmm, I'm not sure.

If the validator was just revoked, technically they still *should* have signed the block in which they were revoked, so I left in the usual accounting logic for that. I think we could safely remove it though.